stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/MessageDispatcher.java

```diff
@@ -196,29 +188,44 @@ private Object getMessageContentForType(byte[] message, BType dataType) throws U
     }
 
     private ObjectValue getMessageObjectValue(byte[] message, long deliveryTag, AMQP.BasicProperties properties) {
-        ObjectValue messageObjectValue = BallerinaValues.createObjectValue(RabbitMQConstants.PACKAGE_ID_RABBITMQ,
-                RabbitMQConstants.MESSAGE_OBJECT);
-        messageObjectValue.addNativeData(RabbitMQConstants.DELIVERY_TAG, deliveryTag);
-        messageObjectValue.addNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT, channel);
-        messageObjectValue.addNativeData(RabbitMQConstants.MESSAGE_CONTENT, message);
-        messageObjectValue.addNativeData(RabbitMQConstants.AUTO_ACK_STATUS, autoAck);
-        if (!Objects.isNull(rabbitMQTransactionContext)) {
-            messageObjectValue.addNativeData(RabbitMQConstants.RABBITMQ_TRANSACTION_CONTEXT,
-                    rabbitMQTransactionContext);
+        try {
+            ObjectValue messageObjectValue = BallerinaValues.createObjectValue(RabbitMQConstants.PACKAGE_ID_RABBITMQ,
+                    RabbitMQConstants.MESSAGE_OBJECT);
+            messageObjectValue.set(RabbitMQConstants.DELIVERY_TAG, deliveryTag);
+            messageObjectValue.set(RabbitMQConstants.JAVA_CLIENT_CHANNEL, new HandleValue(channel));
+            messageObjectValue.set(RabbitMQConstants.MESSAGE_CONTENT, new ArrayValue(message));
+            messageObjectValue.set(RabbitMQConstants.AUTO_ACK_STATUS, autoAck);
+            messageObjectValue.set(RabbitMQConstants.MESSAGE_ACK_STATUS, false);
+            if (properties != null) {
+                String replyTo = properties.getReplyTo();
+                String contentType = properties.getContentType();
+                String contentEncoding = properties.getContentEncoding();
+                String correlationId = properties.getCorrelationId();
+                MapValue<String, Object> basicProperties =
+                        BallerinaValues.createRecordValue(RabbitMQConstants.PACKAGE_ID_RABBITMQ,
+                                RabbitMQConstants.RECORD_BASIC_PROPERTIES);
+                Object[] values = new Object[4];
+                values[0] = replyTo;
+                values[1] = contentType;
+                values[2] = contentEncoding;
+                values[3] = correlationId;
+                messageObjectValue.set(RabbitMQConstants.BASIC_PROPERTIES,
+                        BallerinaValues.createRecord(basicProperties, values));
+            }
+            return messageObjectValue;
+        } catch (Throwable e) {
+            e.printStackTrace();
```

Comment:
Do we need to print stacktrace? This will expose java details to the ballerina user.